content/docs/101-why-effect/index.md

```diff
@@ -0,0 +1,151 @@
+---
+title: Why should I use this?
+---
+
+Effect offers a different approach writing TypeScript than what most people are used to. However, to buy into using Effect, you don't have to use the entire system. You can use pieces of the system independently, and migrate your codebase incrementally without committing to everything at once.
+<br />
+With that said, the rest of this page is dedicated to trying to demonstrate to you that Effect is worth buying into as an entire system for writing TypeScript, not just as a small set of utility functions in random parts of your program.
+<br />
+Let's jump into an example.
+<br />
+For the following set of examples, observe how the structure of the code changes, and don't worry about how we implement a specific action like sending an email.
+<br />
+
+## Simple program (no error handling)
+
+Let's imagine a simple TypeScript program that creates a user in the database and invites a user via email.
+
+```ts twoslash
+declare function createUserInDB(args: { name: string; email: string }): { id: string; name: string; email: string }
```

Comment:
Should `createUserInDB` and `sendUserInvite` return Promises?

Reply:
You're right, given how I'm using it, technically should be a Promise